### PR TITLE
Make check for the limited cmake dependencies the part of sparse checkout

### PR DESCRIPTION
**Pull request by Felixoid**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
A tiny followup for #56898

### CODE REVIEW

File: contrib/update-submodules.sh [hunk 1]
```diff
@@ -9,4 +9,17 @@ cd $GIT_DIR
 contrib/sparse-checkout/setup-sparse-checkout.sh
 git submodule init
 git submodule sync
-git config --file .gitmodules --get-regexp .*path | sed 's/[^ ]* //' | xargs -I _ --max-procs 64 git submodule update --depth=1 --single-branch _
+# NOTE: do not use --remote for `git submodule update`[1] command, since the submodule references to the specific commit SHA1 in the subproject.
+#       It may cause unexpected behavior. Instead you need to commit a new SHA1 for a submodule.
+#
+#       [1] - https://git-scm.com/book/en/v2/Git-Tools-Submodules
+git config --file .gitmodules --get-regexp '.*path' | sed 's/[^ ]* //' | xargs -I _ --max-procs 64 git submodule update --depth=1 --single-branch _
+||||||| parent of 32834adf17d (Make check for the limited cmake dependencies the part of sparse checkout)
```
Comment (antonio2368) [inline contrib/update-submodules.sh:17]: remove?

Comment (Felixoid) [reply contrib/update-submodules.sh]: Yes, rebase leftover

**Review by antonio2368 — APPROVED** [review]

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit c3a3cf8d24d5ca1205c2fe9cb2d8d266ea276b00 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/56991/c3a3cf8d24d5ca1205c2fe9cb2d8d266ea276b00/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/6929871403">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/56991/c3a3cf8d24d5ca1205c2fe9cb2d8d266ea276b00/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

**Comment by Felixoid** [comment]
it worked for both [macos](https://github.com/ClickHouse/ClickHouse/actions/runs/6929871403/job/18849745951?pr=56991#step:8:233) and [package_debug](https://github.com/ClickHouse/ClickHouse/actions/runs/6929871403/job/18849746958?pr=56991#step:8:629) with a clean repo in [other](https://github.com/ClickHouse/ClickHouse/actions/runs/6929871403/job/18849744718?pr=56991#step:8:232) builds

**Pull request merged**